### ISSUE TITLE
using markdown in javadoc comments

### ISSUE THREAD

**Issue by x12a1f** [testingisdocumenting/znai#1040]
I am using a doclet when generating javadocs which allows me to use markdown in the javadoc comments in the .java files. 

Is there an option so that :include-java-doc: processes the javadoc comments as markdown? Currently if I use :include-java-doc: on javadoc formatted with markdown, the result is just an unformatted wall of text.

**Comment by MykolaGolubyev** [comment]
This makes a lot of sense. Thanks for the reporting. 
 No option at the moment and makes sense to add it. 
 
I wonder if znai should automatically treat text as markdown or should detect the absence of html tags. Or maybe a flag.

**Comment by x12a1f** [comment]
I think a global setting to treat javadoc text as markdown would be best. With perhaps a flag to :include-java-doc: to override the global setting.

I think auto detecting would be difficult. Not all valid javadoc have HTML tags but would look weird if rendered using markdown.

**Comment by MykolaGolubyev** [comment]
I like the global settings idea! Right now global and page local plugin settings is still in a backlog and this use case helps to prioritize. 

What do you think if we introduce plugin parameter only for now, something like `markdown: true` and then it can become part of the next release.

The releases after that will get the global settings idea. If also have ideas on how would you like to specify global plugin params - do share!

**Comment by x12a1f** [comment]
Yes, adding a `markdown: true` parameter would be really nice to have.

As for global settings. I think they can be added to meta.json:

```json
{
  "plugins": {
    "java-doc": {
      "markdown": true
    }
  }
}
```

so any settings in meta.json will overwrite the default settings for specific plugin parameters

**Comment by MykolaGolubyev** [comment]
Really like the meta.json plugins section idea!
Will see if I can squeeze it into upcoming release

**Comment by MykolaGolubyev** [comment]
@x12a1f do you plan to use JavaDocs tags like {@link} mixed with markdown?

**Comment by x12a1f** [comment]
I've checked and there are indeed {@link} tags. There are also some {@inheritDoc} but I don't mind if those won't work.

It would be great if javadoc-parameters plugin can also support markdown.

Thanks.

**Comment by MykolaGolubyev** [comment]
#1051

**Comment by MykolaGolubyev** [comment]
this has been released as 1.62: https://testingisdocumenting.org/znai/release-notes/2022#1-62
along with global and page local plugin defaults
